### PR TITLE
Change WalletHome to be a function component

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Changed the `WalletHome` component to be a function component. This is the main screen of the app and was due for the update. No logic should have been changed.

### Other changes

- Changed in some places the wording of `gold` for `celo`. Gold is the original name of the token but was changed a longtime ago. Didn't go through all the code to change it, only the places that were affected by this PR.

### Tested

Manually and updated the unit tests.

### How others should test

Just go through the test plan normally and make sure everything continues to work.

### Related issues

- Part of #546

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/mobile/src/home/WalletHome.tsx [hunk 1]
```diff
@@ -1,212 +1,156 @@
...
+        t('testnetAlert.0', { testnet: _.startCase(DEFAULT_TESTNET) })
+      )
+    )
+  }
 
-    this.scrollPosition = new Animated.Value(0)
-    this.onScroll = Animated.event([{ nativeEvent: { contentOffset: { y: this.scrollPosition } } }])
-    this.state = {
-      isMigrating: false,
+  const tryImportContacts = async () => {
```
Comment (isabellewei) [inline packages/mobile/src/home/WalletHome.tsx:66]: do we no longer want to check if the user's number is verified before importing contacts?

**Review by isabellewei — APPROVED** [review]
LGTM! Thanks for doing this :)

File: packages/mobile/src/home/WalletHome.tsx [hunk 2]
```diff
@@ -1,212 +1,156 @@
...
 
-  shouldShowCashInBottomSheet = () => {
-    const hasStable = STABLE_CURRENCIES.some((currency) =>
-      this.props.balances[currency]?.isGreaterThan(STABLE_TRANSACTION_MIN_AMOUNT)
-    )
-    const hasGold = this.props.balances[Currency.Celo]?.isGreaterThan(GOLD_TRANSACTION_MIN_AMOUNT)
-    const isAccountBalanceZero = !hasStable && !hasGold
+  useEffect(() => {
+    if (appState === AppState.Active) {
+      dispatch(refreshAllBalances())
```
Comment (jeanregisser) [inline packages/mobile/src/home/WalletHome.tsx:92]: not directly related to this PR, but I noticed due to the way the sagas are initialized, this action isn't handled by any saga (still waiting in the blocking `appInit` saga).
But maybe we can remove this soon anyway with the multi token changes?

File: packages/mobile/src/home/WalletHome.test.tsx [hunk 3]
```diff
@@ -1,139 +1,133 @@
...
-    )
-    expect(tree).toMatchSnapshot()
+    const { getByTestId } = renderScreen({
+      ...zeroBalances,
+      app: {
+        cashInButtonExpEnabled: true,
+      },
+    })
+
+    expect(getByTestId('cashInBtn')).toBeTruthy()
```
Comment (jeanregisser) [inline packages/mobile/src/home/WalletHome.test.tsx:120]: Love seeing more useful tests and not just a big snapshot test that nobody looks at ;)

**Review by jeanregisser — APPROVED** [review]
Amazing!! Thanks so much! 🙏 💯

File: packages/mobile/src/home/WalletHome.tsx (hunk 1 above)
Comment (gnardini) [reply packages/mobile/src/home/WalletHome.tsx:66]: Yeah, I don't know why I changed that, didn't mean to, undid that change, thanks!

File: packages/mobile/src/home/WalletHome.tsx (hunk 2 above)
Comment (gnardini) [reply packages/mobile/src/home/WalletHome.tsx:92]: We still need to fetch balances on app load, for now I just added `yield call(refreshBalances)` to the saga initialization, but we should think about how to handle the token fetching in a more robust way. I think a good time to do that is when we remove the old code

**Comment by ValoraQA** [comment]
Hey @MuckT We have verified this issue using Android internal build and iOS test flight build V:1.22.1 &  updated observation in BVT test plan from test quality tool
**BVT test quality tool link:**  https://valora.testquality.com/project/12128/plan/20403/run/153293/result/2581215
**Query/Issues:**
We have already filed below Query/Bugs in the whole application while performing BVT 
- https://github.com/valora-inc/wallet/issues/931
- https://github.com/valora-inc/wallet/issues/1452
- https://github.com/valora-inc/wallet/issues/935
- https://github.com/valora-inc/wallet/issues/1332
**Verified On Devices:** Samsung Galaxy F41(11.0), iPhone 12(15.0)
CC: @gnardini 
Can you please let us know if any other things need to test here ?
Thanks..!!

**Pull request merged**